crates/rust-analyzer/src/to_proto.rs

```diff
@@ -413,12 +413,18 @@ pub(crate) fn signature_help(
 pub(crate) fn inlay_hint(line_index: &LineIndex, inlay_hint: InlayHint) -> lsp_ext::InlayHint {
     lsp_ext::InlayHint {
         label: inlay_hint.label.to_string(),
-        range: range(line_index, inlay_hint.range),
+        position: match inlay_hint.kind {
+            InlayKind::ParameterHint => position(line_index, inlay_hint.range.start()),
+            _ => position(line_index, inlay_hint.range.end()),
+        },
         kind: match inlay_hint.kind {
-            InlayKind::ParameterHint => lsp_ext::InlayKind::ParameterHint,
-            InlayKind::TypeHint => lsp_ext::InlayKind::TypeHint,
-            InlayKind::ChainingHint => lsp_ext::InlayKind::ChainingHint,
+            InlayKind::ParameterHint => Some(lsp_ext::InlayHintKind::PARAMETER),
+            InlayKind::TypeHint => Some(lsp_ext::InlayHintKind::TYPE),
+            InlayKind::ChainingHint => None,
```

Comment:
why do chaining hints not have their own kind? ([CC](https://github.com/emacs-lsp/lsp-mode/pull/3404#discussion_r826485581))

Reply:
Most likely cause the proposal only defines type and parameter hint kinds.

Reply:
Yes, here's a current corresponding definition on VSCode side: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/597a8ab3638d98ab8bfdbb07a6d6afec1dd45ab4/types/vscode/index.d.ts#L4581-L4596

Reply:
I suppose a chaining hint is a kind of type hint?

Reply:
Could be.
If we believe their docs, it should affect the appearance only(?)
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/597a8ab3638d98ab8bfdbb07a6d6afec1dd45ab4/types/vscode/index.d.ts#L4671-L4674